scripts/ci/cut_ironclaw_release.py

```
@@ -166,22 +167,44 @@ def _checked_out_sha(candidate_root: Path) -> str:
-        return str(tomllib.load(manifest_file)["package"]["version"])
+
+    matches: list[tuple[Path, dict[str, object]]] = []
+    for crate_dir in crate_dirs:
+        manifest = candidate_root / crate_dir / "Cargo.toml"
+        try:
+            with manifest.open("rb") as manifest_file:
+                document = tomllib.load(manifest_file)
+        except OSError as error:
+            raise ReleaseTagError(
+                f"cannot read candidate manifest {manifest}: {error}"
+            ) from error
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)